src/components/tools/pen.js

```diff
@@ -0,0 +1,182 @@
+/**
+ * Pen tool
+ * A tool that allows drawing on networked-drawing components.
+ * @namespace drawing
+ * @component pen
+ */
+
+function almostEquals(epsilon, u, v) {
+  return Math.abs(u.x - v.x) < epsilon && Math.abs(u.y - v.y) < epsilon && Math.abs(u.z - v.z) < epsilon;
+}
+
+AFRAME.registerComponent("pen", {
+  schema: {
+    drawFrequency: { default: 5 }, //frequency of polling for drawing points
+    minDistanceBetweenPoints: { default: 0.01 }, //minimum distance to register new drawing point
+    camera: { type: "selector" },
+    drawingManager: { type: "string" },
+    color: { type: "color", default: "#FF0033" },
+    availableColors: {
+      default: [
+        "#FF0033",
+        "#FFFF00",
+        "#0099FF",
+        "#00FF33",
+        "#9900FF",
+        "#FF6600",
+        "#FFFFFF",
+        "#AAAAAA",
+        "#777777",
```

Comment:
probably due to the unlit shader but to actually get a gradient of greys I bet it has to be something like `#000000`, `#111111`, `#222222` or something

also we need brown so i can draw a proper goomba

https://www.mariowiki.com/images/thumb/9/9c/Goomba_-_MarioPartyStarRush.png/200px-Goomba_-_MarioPartyStarRush.png

Reply:
oh, we might want to include one or more flesh tones too

Reply:
so you can draw faces